controller/src/beerocks/master/tasks/association_handling_task.h

```diff
@@ -23,9 +23,9 @@ class association_handling_task : public task {
 
 protected:
     virtual void work() override;
-    virtual void handle_response(std::string radio_mac,
-                                 beerocks_message::eActionOp_CONTROL action_op,
-                                 ieee1905_1::CmduMessageRx &cmdu_rx) override;
+    virtual void handle_response(
+        std::string radio_mac, beerocks_message::eActionOp_CONTROL action_op,
+        std::shared_ptr<beerocks::beerocks_header> beerocks_header) override;
```

Comment:
[CppCoreGuideline R.30](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#r30-take-smart-pointers-as-parameters-only-to-explicitly-express-lifetime-semantics) Use reference not smartptr as function argument.

I think I need to make a CODING_STYLE entry for this...

Reply:
You are right basically but for the time being, I prefer to keep it as a shared_ptr since changing it to reference will cause a massive change to all the callers I would prefer to avoid at this stage.

Reply:
You're *introducing* the shared_ptr here, so I don't see what you mean with the massive change to all callers. I mean, yes it's a massive change to all callers, but you're doing that change anyway.